### ISSUE TITLE
Fix incorrect loading of template files

### ISSUE THREAD

**Issue by adam-buckley** [2pisoftware/cmfive#174]
Currently if I have a structure like so:

- module
  - actions
    - api
      - index.php
    - index.php
  - other standard folders...
  - templates
    - index.tpl.php

If in my api/index.php action I call Web::setLayout(null) but echo my content to screen, the templates/index.tpl.php is loaded even though I explicitly set the layout to null, I expect the echoed content to be printed to screen and that's it. The cause of this issue is (at the time of writing) from line 433-438 of web.php:
``` php
// evaluate template only when buffer is empty
if (sizeof($this->_buffer) == 0) {
    $body = $this->fetchTemplate();
} else {
    $body = $this->_buffer;
}
```

On top of that, the fetchTemplate() function mentioned above also calls templateExists(), which looks in the submodule directory if it exists, then proceeds to look in the module template base directory, the module directory and the project template directory.

``` php
function templateExists($name) {
        if ($this->_submodule) {
            $paths[] = implode("/", array($this->getModuleDir($this->_module), $this->_templatePath, $this->_submodule));
        }
        $paths[] = implode("/", array($this->getModuleDir($this->_module), $this->_templatePath));
        $paths[] = implode("/", array($this->getModuleDir($this->_module)));
        $paths[] = implode("/", array($this->_templatePath, $this->_module));
        $paths[] = $this->_templatePath;
        
        // Add system fallback
        $paths[] = SYSTEM_PATH . "/" . $this->_templatePath;
    ....
```

This makes sense for layout templates because you need a fallback and override mechanism, but not for action templates, this to me is wrong and is producing undesired effects. The work around is to call $w->out(), but I should also be able to safely echo without the system trying to guess what template I want, especially when I've already called Web::setLayout(null).

Adam

**Comment by adam-buckley** [comment]
In addition, a quick fix could be to replace line 434 with
``` php
if (sizeof($this->_buffer) == 0 && !empty($this->_layout)) {
```
But I'm not sure what issues that could produce

**Comment by adam-buckley** [comment]
This issue is also produced because I used `echo` instead of `$w->out()`. Which, depending on how you look at it, is in itself an issue. If I set the layout to null then just echo-ing should be fine but the way cmfive builds the buffer means that by the same token we should _always_ use `$w->out()`. However I don't believe we should enforce that type of usage.

The fix 4cb9d598836ff0012858efc87fd0d102d1e15e20 I have pushed caters for both use cases.
